### DISCUSSION TITLE
Theme-Redoc Failing

### DISCUSSION THREAD

**Discussion by malathikrishnap** [rohit-gohri/redocusaurus#344] [Q&A] [answered]
Hi there,

I'm getting this error while building the project.
_could not render static page with path /docs/APIs/api-intro because of error: Cannot access 'theme_Redoc' before initialization_

I have the following in my .md file:

import ApiDocMdx from '@theme/ApiDocMdx';

<ApiDocMdx id="connector-group-existing" />

Can you please give any pointers to this issue?
It was all working fine until now. Today I cleared my build and nodeModules folders. After rebuilding the project, I'm getting this issue.

**Comment by malathikrishnap** [comment]
Any quick help on this is highly appreciated.
I noticed that you modified the package 5 days ago. Hope that didn't cause any issues for me. just curious.

**Comment by rohit-gohri** [comment]
Hi @malathikrishnap, have you setup the config with the spec according to docs - https://redocusaurus.vercel.app/docs/getting-started/Installation 

Could you share a reproduction? The latest release is tested and is working

**Reply by malathikrishnap** [reply]
Thanks for the revert, Rohit.
My project was all fine and running well until today morning.
I just deleted the build and nodeModules, and rebuilt. It didn't work.

Did anything change in the installation commands in the past few days?

**Reply by malathikrishnap** [reply]
presets: [
    [
      'classic',
      /** @type {import('@docusaurus/preset-classic').Options} */
      ({
        docs: {
          sidebarPath: './sidebars.js',
        },
        blog: {
          showReadingTime: true,
        },
        theme: {
          customCss: './src/css/custom.css',
        },
      }),
    ],
    [
      'redocusaurus',
      {
        // Plugin Options for loading OpenAPI files
        config: path.join(__dirname, 'redocly.yaml'),
        specs: [
          {
            id: 'create-connector-group',
            spec: 'docs/APIs/connectorsOpenAPISpec.json',
            route: '/api/connectorgroup',
          },
          {
            id: 'connector-group-existing',
            spec: 'openAPI.json',
            route: '/api/connectorgroupexisting',
          },
        ],
        // Theme Options for modifying how redoc renders them
        theme: {
          // Change with your site colors
          primaryColor: '#1890ff',
          options: { disableSearch: false },
          themes: ['docusaurus-theme-redoc'],
        },
      },
    ],
  ],

**Reply by malathikrishnap** [reply]
Here's the complete error stack
[RedocError.pdf](https://github.com/rohit-gohri/redocusaurus/files/14584227/RedocError.pdf)

**Reply by rohit-gohri** [reply]
Could you try downgrading to [2.0.1](https://www.npmjs.com/package/redocusaurus/v/2.0.1)? I didn't change anything related to your error so I'm not sure but would still like to check with older version

**Reply by malathikrishnap** [reply]
I downgraded both redocusaurus and the docusaurus-theme-redoc to 2.0.1
Still I get the same error :(
[RedocError.pdf](https://github.com/rohit-gohri/redocusaurus/files/14584599/RedocError.pdf)

**Reply by rohit-gohri** [reply]
Can you remove `themes: ['docusaurus-theme-redoc'],` from your config? You don't need the theme directly if you are already using the preset

**Reply by malathikrishnap** [reply]
I did that, still the same.

**Reply by malathikrishnap** [reply]
I tried these with no luck:

- npm install docusaurus-theme-redoc
- npm i -save docusaurus-theme-redoc
- npm i -save redocusaurus

<img width="364" alt="Screenshot 2024-03-13 at 2 11 50 PM" src="https://github.com/rohit-gohri/redocusaurus/assets/112460089/426d5f4c-d406-4237-b626-d444ceb58fc7">
These are the folder contents:

**Reply by rohit-gohri** [reply]
Can you completely remove `docusaurus-theme-redoc`, it's not meant to be used directly.

**Reply by rohit-gohri** [reply]
If it doesn't work then I can't really help much without a proper reproduction

**Reply by malathikrishnap** [reply]
I found the culprit. I added this Prism snippet to docusaurus.config.js two days back.
prism: {
          theme: prismThemes.github,
          darkTheme: prismThemes.dracula,
          **_additionalLanguages: ['java'],_**
        },

I added "Java" as the additional language to Prism to highlight my java code. Never knew this could throw an error on the redoc theme.
I removed this line and the API spec is loading fine.

Rohit, thanks a ton for all your time and patience!

**Reply by rohit-gohri** [reply, marked as answer]
There is a workaround for that - https://github.com/rohit-gohri/redocusaurus/issues/68#issuecomment-877902022

**Reply by malathikrishnap** [reply]
Yes, the workaround is working. Thanks for sharing that!
